compiler/packages/babel-plugin-react-compiler/src/HIR/Environment.ts

```diff
@@ -331,6 +331,21 @@ const EnvironmentConfigSchema = z.object({
    */
   enableInstructionReordering: z.boolean().default(false),
 
+  /**
+   * If a given component has a single return statement and returns JSX where the
+   * tag is known (via moduleTypeProvider) to be a function component, then it is
+   * safe to convert the JSX expression into a direct call to the component in question:
+   *
+   * function Parent() {
+   *   return <Child />
+   * }
+   * =>
+   * function Parent() {
+   *   return Child({})
+   * }
+   */
```

Comment:
This is the basic idea. We have to know that the tag corresponds to a function component, though, for this transformation to be safe.